'big.js': EISDIR: illegal operation on a directory, read

When trying to import [big.js](https://github.com/MikeMcl/big.js/) package, TypeScript attempts to read it as file, resulting in EISDIR error.

For repro, create the following tsconfig.json:
```json
{
  "compilerOptions": {
    "allowJs": true,
    "outDir": "lib",
    "module": "commonjs",
    "target": "es5",
    "baseUrl": ".",
    "paths": {
      "*": [
        "*", 
        "node_modules/*"
      ]
    }
  },
  "files": ["test.js"]
}
```

and source file:
```javascript
import 'big.js';
```

And execute:
```sh
npm install big.js
tsc
```

The result:
```log
test.js(1,8): error TS5012: Cannot read file 'C:/Test/ts-eisdir/node_modules/big.js': EISDIR: illegal operation on a directory, read
```

The [npm package name rules](https://docs.npmjs.com/files/package.json) say the name can't start with a period, but it seems it's allowed to contain a period. And although it's recommended not to have "js" in the name, it's not disallowed.

--- a/src/compiler/sys.ts
+++ b/src/compiler/sys.ts
@@ -411,7 +411,7 @@ namespace ts {
             const useCaseSensitiveFileNames = platform !== "win32" && platform !== "win64" && platform !== "darwin";
 
             function readFile(fileName: string, encoding?: string): string {
-                if (!_fs.existsSync(fileName)) {
+                if (!fileExists(fileName)) {
                     return undefined;
                 }
                 const buffer = _fs.readFileSync(fileName);
@@ -460,6 +460,31 @@ namespace ts {
 
             function getCanonicalPath(path: string): string {
                 return useCaseSensitiveFileNames ? path : path.toLowerCase();
+            }
+
+            const enum FileSystemEntryKind {
+                File,
+                Directory
+            }
+            function fileSystemEntryExists(path: string, entryKind: FileSystemEntryKind): boolean {
+                try {
+                    const stat = _fs.statSync(path);
+                    switch (entryKind) {
+                        case FileSystemEntryKind.File: return stat.isFile();
+                        case FileSystemEntryKind.Directory: return stat.isDirectory();
+                    }
+                }
+                catch (e) {
+                    return false;
+                }
+            }
+
+            function fileExists(path: string): boolean {
+                return fileSystemEntryExists(path, FileSystemEntryKind.File);
+            }
+
+            function directoryExists(path: string): boolean {
+                return fileSystemEntryExists(path, FileSystemEntryKind.Directory);
             }
 
             function readDirectory(path: string, extension?: string, exclude?: string[]): string[] {
@@ -538,12 +563,8 @@ namespace ts {
                 resolvePath: function (path: string): string {
                     return _path.resolve(path);
                 },
-                fileExists(path: string): boolean {
-                    return _fs.existsSync(path);
-                },
-                directoryExists(path: string) {
-                    return _fs.existsSync(path) && _fs.statSync(path).isDirectory();
-                },
+                fileExists,
+                directoryExists,
                 createDirectory(directoryName: string) {
                     if (!this.directoryExists(directoryName)) {
                         _fs.mkdirSync(directoryName);

--- a/src/compiler/sys.ts
+++ b/src/compiler/sys.ts
@@ -466,6 +466,7 @@ namespace ts {
                 File,
                 Directory
             }
+
             function fileSystemEntryExists(path: string, entryKind: FileSystemEntryKind): boolean {
                 try {
                     const stat = _fs.statSync(path);

